SwiftCompilerSources/Sources/AST/Declarations.swift

```diff
@@ -165,3 +176,150 @@ extension Optional where Wrapped == Decl {
     OptionalBridgedDeclObj(self?.bridged.obj)
   }
 }
+
+public typealias Identifier = swift.Identifier
+
+public typealias GenericTypeParamKind = swift.GenericTypeParamKind
+
+public typealias ASTContext = BridgedASTContext
+
+public typealias DeclContext = BridgedDeclContext
+
+public typealias Expr = BridgedExpr
+
+public typealias ParameterList = BridgedParameterList
+
+public typealias SourceFile = BridgedSourceFile
+
+public typealias FileUnit = BridgedFileUnit
+
+public typealias GenericParamList = BridgedGenericParamList
+
+public typealias TrailingWhereClause = BridgedTrailingWhereClause
+
+public typealias BridgedParamDecl = ASTBridging.BridgedParamDecl
+
+public typealias BridgedGenericTypeParamDecl = ASTBridging.BridgedGenericTypeParamDecl
+
+public typealias BridgedEnumDecl = ASTBridging.BridgedEnumDecl
+
+public typealias BridgedEnumElementDecl = ASTBridging.BridgedEnumElementDecl
+
+extension ParameterList {
+  public subscript(_ index: Int) -> BridgedParamDecl {
+    return get(index)
+  }
+
+  public static func createParsed(
+    _ astContext: ASTContext, leftParenLoc: SourceLoc?, parameters: [BridgedParamDecl],
+    rightParenLoc: SourceLoc?
+  ) -> ParameterList {
+    parameters.withBridgedArrayRef {
+      ParameterList.createParsed(
+        astContext, leftParenLoc: leftParenLoc.bridgedLocation, parameters: $0,
+        rightParenLoc: rightParenLoc.bridgedLocation)
+    }
+  }
+}
+
+extension GenericParamList {
+  public static func createParsed(
+    _ astContext: ASTContext, leftAngleLoc: SourceLoc?, parameters: [BridgedGenericTypeParamDecl],
+    genericWhereClause: TrailingWhereClause?,
+    rightAngleLoc: SourceLoc?
+  ) -> GenericParamList {
+    return parameters.withBridgedArrayRef {
+      GenericParamList.createParsed(
+        astContext, leftAngleLoc: leftAngleLoc.bridgedLocation, parameters: $0,
+        genericWhereClause: genericWhereClause.bridged, rightAngleLoc: rightAngleLoc.bridgedLocation
+      )
+    }
+  }
+}
+
+extension BridgedDecl {
+  public var declObj: BridgedDeclObj {
+    BridgedDeclObj(SwiftObject(raw.bindMemory(to: BridgedSwiftObject.self, capacity: 1)))
+  }
+  public var decl: Decl { declObj.decl }
+}
+
+extension BridgedEnumDecl {
+  public static func createParsed(
```

Comment:
Put those "create" functions into the real decl classes. E.g.
```
final public class EnumDecl: NominalTypeDecl {
  public static create(...) -> EnumDecl
```

(no need for the suffix "Parsed", unless I'm missing something)

You can easily convert from a `BridgedEnumDecl` to a `BridgedDeclObj`:
```
  BridgedDeclObj::createEnumDecl(...) {
    BridgedEnumDecl ed = BridgedEnumDecl.createParsed(...)
    return ed.unbridged();
  }
```

and on the Swift side:
```
  public static create(...) {
    return BridgedDeclObj.createEnumDecl(...).getAs(EnumDecl.self)
  }
```

Reply:
In order to avoid cluttering `BridgedDeclObj` with many `createXXX` methods, I've decided to directly call `create` static methods of `BridgedXXXDecl` classes, and then cast results to swift-native `XXXDecl` classes. Please let me know if this is applicable.